File: Remove Duplicates from Sorted Array II - M/sol.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i = 0, j = 1, m = 1;
        while (j < nums.size()){
            if (nums[i] == nums[j] && m < 2) {
                i++;
                swap(nums[i],nums[j]);
                m++;
            }
            else if (nums[i] != nums[j] && m >= 2) {
                i++;
                swap(nums[i],nums[j]);
                m = 1;
            }
            else if (nums[i] != nums[j] && m < 2) {
                i++;
                swap(nums[i],nums[j]);
            }
            j++;
        }
        if (m == 2 && nums[i] < nums[j-1]) {
            swap(nums[i],nums[j-1]);
        }       
        return i+1;
    }
};

File: Jump Game - M/sol.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        i = n - 2
        j = n - 1
        
        while i >= 0:
            if nums[i] + i >= j:
                j = i
            i -= 1
        return j == 0
